src/dos/dos_programs.cpp

```diff
@@ -147,12 +148,12 @@ class MOUNT : public Program {
 
 			/* Change 8.3 to 11.0 */
 			char* dot = strchr(name,'.');
-			if(dot && (dot - name == 8) ) { 
+			if(dot && (dot - name == 8) ) {
 				name[8] = name[9];name[9] = name[10];name[10] = name[11];name[11] = 0;
 			}
```

Comment:
(just a comment, but maybe there's a bug here) The more I look at this code, the more I think the original code in here was incorrect…

Normally, these lines should convert `abcdabcd.xyz` to `abcdabcdxyz`, right? But what about `abc.abcd.xyz`? This algorithm won't convert it, but perhaps it should be changed into `abc.abcdxyz`?

BTW, at least let's fix formatting of the whole if statement when touching this ;)

Reply:
Looks like this code prints the drive label, where **FAT volumes have the following limitations:**
- Contain as many as 11 character bytes and can include spaces, but no tabs.
- Volume labels cannot contain the following characters: ? / \ | . , ; : + = [ ] < > "
- Volume labels are stored as upper-case regardless of whether they contain lower-case letters.

Ref: https://en.wikipedia.org/wiki/Label_(command)

Reply:
Addressed it as much as possible here: https://github.com/dreamer/dosbox-staging/commit/f9c8d719d043eaaba1127e3d4b2f8a1cd45895fd

This only performs a read-only cleanup of the label as-printed for the user when they run the 'mount' command.
The label itself is set in the 'Set_Label' function, which could be improved, but out-of-scope of this PR.

Here's a comparison of labels (as printed with `mount`) between SVN and this PR:

![2020-01-19_21-52_1](https://user-images.githubusercontent.com/1557255/72702179-9f7f8b00-3b06-11ea-976d-2a90e0c69eec.png)
![2020-01-19_21-52](https://user-images.githubusercontent.com/1557255/72702181-9f7f8b00-3b06-11ea-82fe-2e89a1d37f33.png)
![2020-01-19_21-45_1](https://user-images.githubusercontent.com/1557255/72702182-9f7f8b00-3b06-11ea-97f3-4f03d78f0fce.png)
![2020-01-19_21-45](https://user-images.githubusercontent.com/1557255/72702183-9f7f8b00-3b06-11ea-9a2a-66824edb3c09.png)

Reply:
Looks so much better, thank you for thorough follow up and implementation :)